### ISSUE TITLE
Fix highlighting of subfolder menu items when moving mail

### ISSUE THREAD

**Issue by macdeport** [RealRaven2000/QuickFolders#88]
When drag and drop of mail on tab I have experienced some errors in selecting the right sub-folder due to the lack of feedback and the too fast (for me at least) disappearance of the sub-folders display after selection... or tentative of selection. Feedback if feasible will improve ergonomic of drag and drop of mail.

**Comment by RealRaven2000** [comment]
> When drag and drop of mail on tab I have experienced some errors in selecting the right sub-folder due to the lack of feedback 

"Lack of feedback" - should actually be fixed in version [5.1](https://github.com/RealRaven2000/QuickFolders/releases/tag/5.1) that was handled in issue #79 

> and the too fast (for me at least) disappearance of the sub-folders display after selection... or tentative of selection. Feedback if feasible will improve ergonomic of drag and drop of mail.

The dropdown is only removed if you drag the mail away from the tab. You are expected to drag down into the subfolder menu.

**Comment by macdeport** [comment]
I am already in your **Flat Style Theme** and  will check drag and drop anew tomorrow.

**Comment by macdeport** [comment]
The theme selected:
"Bling My Tabs!" / Themes: Flat Style / Individual Tab Color Appearance [Mozilla] (filled style)

With my configuration (Mac OS X) I confirm that the sub-folder menu disappears immediately when a sub-folder is selected so if the feedback is in this menu I can't see it :-(

**Comment by RealRaven2000** [comment]
> With my configuration (Mac OS X) I confirm that the sub-folder menu disappears immediately when a sub-folder is selected so if the feedback is in this menu I can't see it :-(

maybe i don't understand, but when you click on a  item (essentially a folder) Thunderbird is supposed to jump to that folder - and the menu will close.  if you drag an email there the mail will moved there. what other behavior are you expecting?

**Comment by macdeport** [comment]
> maybe i don't understand,
Maybe I don't explain clearly ;-)

I am moving a mail from my current folder to a sub-folder present in QF bar.

So I drag and drop this selected mail on a given tab witch nicely open a menu of sub-folders...

> try this: drag the mail to a quickfolder tab - don't let go of the mouse button, the subfolders menu should appear immediately - keep dragging until you are over a subfolder. then let go of the mouse button. what happens?

All things are perfectly OK the mail move in the right sub-folder.

But I just suggest some "blinking" (highlighting) as acknowledgement of the right job (intended  sub-folder actually selected) done...

**Comment by RealRaven2000** [comment]
My previous answer was done from my mobile phone, so I couldn't include screenshots. You're not supposed to let go of the mouse button when you are over the tab. That will move the folder to the folder, right away. It's not supposed to open a menu.

(1) moving to a folder that is represented by a tab:
![image](https://user-images.githubusercontent.com/6314674/102689432-fa79a080-41f5-11eb-8e84-501ea69b0c2d.png)

(2) moving to subfolder SeaWoney/test - only let go of the mouse button once on the subfolder "test", not before:
![image](https://user-images.githubusercontent.com/6314674/102689610-63154d00-41f7-11eb-92ef-58551f9faa59.png)

(3) moving to a folder within the subfolder "SeaWonkey/test/mütt" - don't let go of the mouse button until your on the sub/subfolder:
![image](https://user-images.githubusercontent.com/6314674/102689476-5512fc80-41f6-11eb-9243-f4cb2ad03f0f.png)

you only let go of the mouse button once you are over the destination folder - drag to tab, then down, then right, then let go of the mouse button.

Alternatively, to move mail anywhere at all, you can move to the quickMove button and let go. Then enter Se/te/m to get to any folder:

![image](https://user-images.githubusercontent.com/6314674/102689543-d7032580-41f6-11eb-8762-2b65818ccea2.png)

In this case the folder is "inbox" (the tab seaWonkey points to it, apparently I haven't coded using the alias with subfolders, so that's something I can and should add)
![image](https://user-images.githubusercontent.com/6314674/102689563-00bc4c80-41f7-11eb-8cac-f7f83f490fe2.png)

This can also be triggered with the shortcut Shift+M (Move) if you have a QuickFolders Pro license
The pattern Inb/t/ tells QuickFolders to look for a parent folder starting with "Inb" and a child folder starting with "t", in this case Inbox/test

**Comment by RealRaven2000** [comment]
> > try this: drag the mail to a quickfolder tab - don't let go of the mouse button, the subfolders menu should appear immediately - keep dragging until you are over a subfolder. then let go of the mouse button. what happens?

> All things are perfectly OK the mail move in the right sub-folder.

> But I just suggest some "blinking" (highlighting) as acknowledgement of the right job (intended sub-folder actually selected) done...

ok, so in the Thunderbird 68 version we had an icon which was styled using the CSS pseudoselector `-moz-drag-over`. Looked like this:
![image](https://user-images.githubusercontent.com/6314674/102689721-36ae0080-41f8-11eb-923c-3d5326d212f6.png)

the problem is that Mozilla in it's infinite wisdom removed that CSS selector (they claim due to not much usage by web developers. Which is understandable because webpage designers don't  really use drag-and-drop). That's why I had to recode my tabs to attach a special css class for the drag-over feedback (maintenance version 5.1). Obviously I need to do the same with all menu items. Fun and games...

**Comment by RealRaven2000** [comment]
I have retitled the issue and added the label "Tb 78 regression", because this is what it is. Extra work imposed by the latest version of Thunderbird for something that was working perfectly well in all previous versions.

**Comment by macdeport** [comment]
- Thanks for : your care of users, your very clear and detailed explanation, your share of your development _Fun and games_
- I suggest to add your detailed private help about move and tabs (already written) in the general QF help for some new user... if not yet done ;-)

**Comment by RealRaven2000** [comment]
> * Thanks for : your care of users, your very clear and detailed explanation, your share of your development _Fun and games_
> * I suggest to add your detailed private help about move and tabs (already written) in the general QF help for some new user... if not yet done ;-)

Interesting newsflash on this particular issue, it kind of prompted me to leave some feedback on the Thunderbird developers group on the matrix chat (replacing their IRC group) - and found out that `-moz-drag-over` should actually not be removed / deprecated. So I added a new comment to this bug:
https://bugzilla.mozilla.org/show_bug.cgi?id=1659232

If we can create a regression test for this (once I figure out how to do that) they might have a look at what the underlying issue is in Thunderbird - maybe I won't even have to fix it on my side. I will look at the `:-moz-drag-over` rules I have removed and see what happens if I bring them back. As regards demoing how this works I have a few videos on my YouTube Channel here:

https://www.youtube.com/watch?v=ZAu5hyoyLEA&list=PLApv7QYQO9nR_ySMlAYd_wlhei-MRND89

I wanted to get some more videos up with the help of a friend in USA but he has been flat out himself, so not sure if it's going to happen in 2020.

**Comment by macdeport** [comment]
Appreciate Santa Claus solution if confirmed ;-)

**Comment by RealRaven2000** [comment]
After some back and forth [on Bugzilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1659232) I think I may have found a solution. First test version:

[QuickFolders-wx-5.3pre9.zip](https://github.com/RealRaven2000/QuickFolders/files/5873238/QuickFolders-wx-5.3pre9.zip)

the main change was the way I attached the event to the menu item; instead of using the "ondragenter" attribute, like this:
`menuitem.setAttribute("ondragenter","event.preventDefault();")`

Instead, I am now using

```
  menuitem.addEventListener("dragenter", function(e) { 
    e.preventDefault(); 
  });
```
One theory is that the "ondragenter" event attribute is not supported anymore. As I am using quite a bit of event attributes (instead of adding my own listeners) changes like this can be quite disruptive and may cause substantial refactoring work.

**Comment by RealRaven2000** [comment]
One more quick fix - I didn't like the fact that menu items in a subfolder were underlines when we drag over the parent folder:

![image](https://user-images.githubusercontent.com/6314674/105847268-4aa60880-5fd5-11eb-8542-70eff49eb179.png)

So I made some improvements to the CSS selector to only underline the currently hovered item (and no children) even when a menu is dragged over (Will need to backport this fix to Tb68 as well.):

[QuickFolders-wx-5.3pre13.zip](https://github.com/RealRaven2000/QuickFolders/files/5873347/QuickFolders-wx-5.3pre13.zip)

**Comment by RealRaven2000** [comment]
Fixed in 5.3 - released 22/02/2021
